### DISCUSSION TITLE
LPRM AMSRE soil moisture assimilation

### DISCUSSION THREAD

**Discussion by nchawang** [NASA-LIS/LISF#1120] [Q&A]
Hi LIS community,

I was looking to assimilate the LPRM AMSR-E soil moisture.

I had downloaded the LPRM AMSR-E soil moisture data from Descending (https://disc.gsfc.nasa.gov/datasets/LPRM_AMSRE_D_SOILM3_002/summary) and Ascending (https://disc.gsfc.nasa.gov/datasets/LPRM_AMSRE_A_SOILM3_002/summary?keywords=LPRM_AMSRE_A_SOILM3_002). When running the CDF generation i get the following error: 

_"[ERR] Error nf90_inq_varid: RFI_Daily Stopping.                   -49
endrun is being called"_

While checking the variables in the netCDF file and from the above links I realize the variable is not present, while the code, "ldt/DAobs/LPRM_AMSREsm/read_LPRM_AMSREsm.F90" Line 189 calls for looking for the RFI_Daily variable. Am i supposed to download the LPRM AMSR-E nc file from somewhere else did i download it from the wrong version? (It was from "lis/datassim/obs/LPRM_AMSREsm/LPRM_AMSREsm_Mod.F90" under revision that pointed towards NETCDF version from GES-DISC.)

Additionally, ldt/DAobs/LPRM_AMSREsm/read_LPRM_AMSREsm.F90 searches for variables : Rainfall Diagnostic, sm flags and sm combined which are also not present in the netCDF file from GES DISC.

Any help will be highly appreciated.
Thankyou!

**Comment by LIS-navari** [comment]
Hi, 

I believe we developed LPRM AMSR-E soil moisture data assimilation application several years ago, and it has not been updated since then. 

The naming convention of the file we used is 
AMSR_L3_LPRMv05_D_20110929T000000_20110929T235959D.v05.nc

But, the naming convention of the latest version of data is 
LPRM-AMSR_E_L3_D_SOILM3_V002_20111002.nc

The old file contains the following variables: 
ncdump -h LPRM.v5/2011-09/AMSR_L3_LPRMv05_D_20110929T000000_20110929T235959D.v05.nc
netcdf AMSR_L3_LPRMv05_D_20110929T000000_20110929T235959D.v05 {
dimensions:
	Xtrack = 720 ;
	Track = 1440 ;
variables:
	double lon(Xtrack, Track) ;
		lon:long_name = "longitude" ;
		lon:units = "degrees_east" ;
		lon:standard_name = "longitude" ;
		lon:actual_range = -179.875, 179.875 ;
		lon:coordinates = "lat lon" ;
		lon:grid_mapping = "wgs84" ;
	double lat(Xtrack, Track) ;
		lat:long_name = "latitude" ;
		lat:units = "degrees_north" ;
		lat:standard_name = "latitude" ;
		lat:actual_range = -89.875, 89.875 ;
		lat:coordinates = "lat lon" ;
		lat:grid_mapping = "wgs84" ;
	int wgs84 ;
		wgs84:name = "WGS 84" ;
		wgs84:epsg = 4326. ;
		wgs84:grid_mapping_name = "latitude_longitude" ;
		wgs84:semi_major_axis = 6378137. ;
		wgs84:semi_minor_axis = 6356752.31424783 ;
		wgs84:inverse_flattening = 298.2572236 ;
		wgs84:comment = "value is equal to EPSG code" ;
	float Soil_Moisture_from_C_band(Xtrack, Track) ;
		Soil_Moisture_from_C_band:long_name = "AMSRv05 Soil Moisture (6.9GHz)" ;
		Soil_Moisture_from_C_band:units = "[m3 m-3]" ;
		Soil_Moisture_from_C_band:_FillValue = -9999.f ;
		Soil_Moisture_from_C_band:actual_range = -99.f, 99.f ;
		Soil_Moisture_from_C_band:coordinates = "lat lon" ;
		Soil_Moisture_from_C_band:grid_mapping = "wgs84" ;
		Soil_Moisture_from_C_band:gain = "0.01" ;
	float Soil_Moisture_Error_from_C_band(Xtrack, Track) ;
		Soil_Moisture_Error_from_C_band:long_name = "AMSRv05 Soil Moisture Error (6.9GHz)" ;
		Soil_Moisture_Error_from_C_band:units = "[m3 m-3]" ;
		Soil_Moisture_Error_from_C_band:_FillValue = -9999.f ;
		Soil_Moisture_Error_from_C_band:actual_range = -99.f, 50.f ;
		Soil_Moisture_Error_from_C_band:coordinates = "lat lon" ;
		Soil_Moisture_Error_from_C_band:grid_mapping = "wgs84" ;
		Soil_Moisture_Error_from_C_band:gain = "0.01" ;
	float Optical_Depth_from_C_band(Xtrack, Track) ;
		Optical_Depth_from_C_band:long_name = "AMSRv05 Optical Depth (6.9GHz)" ;
		Optical_Depth_from_C_band:units = "[-]" ;
		Optical_Depth_from_C_band:_FillValue = -9999.f ;
		Optical_Depth_from_C_band:actual_range = -99.f, 127.f ;
		Optical_Depth_from_C_band:coordinates = "lat lon" ;
		Optical_Depth_from_C_band:grid_mapping = "wgs84" ;
		Optical_Depth_from_C_band:gain = "0.01" ;
	float Soil_Moisture_from_X_band(Xtrack, Track) ;
		Soil_Moisture_from_X_band:long_name = "AMSRv05 Soil Moisture (10.7GHz)" ;
		Soil_Moisture_from_X_band:units = "[m3 m-3]" ;
		Soil_Moisture_from_X_band:_FillValue = -9999.f ;
		Soil_Moisture_from_X_band:actual_range = -99.f, 99.f ;
		Soil_Moisture_from_X_band:coordinates = "lat lon" ;
		Soil_Moisture_from_X_band:grid_mapping = "wgs84" ;
		Soil_Moisture_from_X_band:gain = "0.01" ;
	float Soil_Moisture_Error_from_X_band(Xtrack, Track) ;
		Soil_Moisture_Error_from_X_band:long_name = "AMSRv05 Soil Moisture Error (10.7GHz)" ;
		Soil_Moisture_Error_from_X_band:units = "[m3 m-3]" ;
		Soil_Moisture_Error_from_X_band:_FillValue = -9999.f ;
		Soil_Moisture_Error_from_X_band:actual_range = -99.f, 50.f ;
		Soil_Moisture_Error_from_X_band:coordinates = "lat lon" ;
		Soil_Moisture_Error_from_X_band:grid_mapping = "wgs84" ;
		Soil_Moisture_Error_from_X_band:gain = "0.01" ;
	float Optical_Depth_from_X_band(Xtrack, Track) ;
		Optical_Depth_from_X_band:long_name = "AMSRv05 Optical Depth (10.7GHz)" ;
		Optical_Depth_from_X_band:units = "[-]" ;
		Optical_Depth_from_X_band:_FillValue = -9999.f ;
		Optical_Depth_from_X_band:actual_range = -99.f, 127.f ;
		Optical_Depth_from_X_band:coordinates = "lat lon" ;
		Optical_Depth_from_X_band:grid_mapping = "wgs84" ;
		Optical_Depth_from_X_band:gain = "0.01" ;
	float Land_Surface_Temperature(Xtrack, Track) ;
		Land_Surface_Temperature:long_name = "AMSRv05 Land Surface Temperature (36.5GHz)" ;
		Land_Surface_Temperature:units = "[K]" ;
		Land_Surface_Temperature:_FillValue = -9999.f ;
		Land_Surface_Temperature:actual_range = -99.f, 3184.f ;
		Land_Surface_Temperature:coordinates = "lat lon" ;
		Land_Surface_Temperature:grid_mapping = "wgs84" ;
		Land_Surface_Temperature:gain = "0.01" ;
	float Rainfall_Diagnostic(Xtrack, Track) ;
		Rainfall_Diagnostic:long_name = "AMSRv05 Rainfall diagnostic (36.5GHz-Hor)" ;
		Rainfall_Diagnostic:units = "[-]" ;
		Rainfall_Diagnostic:_FillValue = -9999.f ;
		Rainfall_Diagnostic:actual_range = 0.f, 1.f ;
		Rainfall_Diagnostic:coordinates = "lat lon" ;
		Rainfall_Diagnostic:grid_mapping = "wgs84" ;
		Rainfall_Diagnostic:gain = "[-]" ;
	float SM_Combined(Xtrack, Track) ;
		SM_Combined:long_name = "AMSRv05 Soil Moisture Combined (6.9 & 10.7GHz)" ;
		SM_Combined:units = "[m3 m-3]" ;
		SM_Combined:_FillValue = -9999.f ;
		SM_Combined:actual_range = -99.f, 99.f ;
		SM_Combined:coordinates = "lat lon" ;
		SM_Combined:grid_mapping = "wgs84" ;
		SM_Combined:gain = "0.01" ;
	float SM_Flags(Xtrack, Track) ;
		SM_Flags:long_name = "AMSRv05 Flags" ;
		SM_Flags:units = "[-]" ;
		SM_Flags:_FillValue = -9999.f ;
		SM_Flags:actual_range = -99.f, 10.f ;
		SM_Flags:coordinates = "lat lon" ;
		SM_Flags:grid_mapping = "wgs84" ;
		SM_Flags:gain = "Flags" ;
	float SM_CDF(Xtrack, Track) ;
		SM_CDF:long_name = "AMSRv05 Soil Moisture CDF-matched" ;
		SM_CDF:units = "[m3 m-3]" ;
		SM_CDF:_FillValue = -9999.f ;
		SM_CDF:actual_range = -99.f, 85.0638f ;
		SM_CDF:coordinates = "lat lon" ;
		SM_CDF:grid_mapping = "wgs84" ;
		SM_CDF:gain = "0.01" ;
	float RFI_Daily(Xtrack, Track) ;
		RFI_Daily:long_name = "AMSRv05 RFI index (Daily)" ;
		RFI_Daily:units = "[index]" ;
		RFI_Daily:_FillValue = -9999.f ;
		RFI_Daily:actual_range = 0.f, 4.f ;
		RFI_Daily:coordinates = "lat lon" ;
		RFI_Daily:grid_mapping = "wgs84" ;
		RFI_Daily:gain = "[-]" ;
	float RFI_Monthly(Xtrack, Track) ;
		RFI_Monthly:long_name = "AMSRv05 RFI index (Monthly)" ;
		RFI_Monthly:units = "[mask]" ;
		RFI_Monthly:_FillValue = -9999.f ;
		RFI_Monthly:actual_range = 0.f, 1.f ;
		RFI_Monthly:coordinates = "lat lon" ;
		RFI_Monthly:grid_mapping = "wgs84" ;
		RFI_Monthly:gain = "[-]" ;
	float RFI_Total(Xtrack, Track) ;
		RFI_Total:long_name = "AMSRv05 RFI index (2002-2011)" ;
		RFI_Total:units = "[counts]" ;
		RFI_Total:_FillValue = -9999.f ;
		RFI_Total:actual_range = 0.f, 6685.f ;
		RFI_Total:coordinates = "lat lon" ;
		RFI_Total:grid_mapping = "wgs84" ;
		RFI_Total:gain = "[-]" ;

// global attributes:
		:title = "Land Parameter Retrievals from AMSR-E observations" ;
		:institution = "VU University Amsterdam" ;
		:source = "LPRM" ;
		:history = "V01; v02; v03; v04" ;
		:references = "Owe et al., 2008; Multi-sensor historical climatology of satellite-derived global land surface moisture" ;
		:email = "R.A.M.de.Jeu@vu.nl or R.M.Parinussa@vu.nl" ;
		:comment = "Most recent update: January 2012" ;
		:version = "v05" ;
		:Conventions = "CF-1.4" ;
		:CF\:featureType = "Grid" ;
		:terms_for_use = "These data can be used freely for research purposes provided that the following source is acknowledged: VU University Amsterdam" ;
		:disclaimer = "This data is made available in the hope that it will be useful, but WITHOUT ANY WARRANTY" ;


But the latest version contains these variables 
ncdump -h LPRM-AMSR_E_L3_D_SOILM3_V002_20111002.nc
netcdf LPRM-AMSR_E_L3_D_SOILM3_V002_20111002 {
dimensions:
	Longitude = 1440 ;
	Latitude = 720 ;
variables:
	float Latitude(Latitude) ;
		Latitude:units = "degrees_north" ;
		Latitude:scale_factor = 1.f ;
		Latitude:add_offset = 0.f ;
		Latitude:long_name = "Latitude" ;
		Latitude:_CoordinateAxisType = "Lat" ;
	float Longitude(Longitude) ;
		Longitude:units = "degrees_east" ;
		Longitude:scale_factor = 1.f ;
		Longitude:add_offset = 0.f ;
		Longitude:long_name = "Longitude" ;
		Longitude:_CoordinateAxisType = "Lon" ;
	short ts(Longitude, Latitude) ;
		ts:long_name = "Skin Temperature (2mm)" ;
		ts:units = "Kelvin" ;
		ts:coordinates = "Latitude Longitude" ;
		ts:_FillValue = -32767s ;
		ts:scale_factor = 0.1f ;
		ts:add_offset = 0.f ;
	short soil_moisture_c(Longitude, Latitude) ;
		soil_moisture_c:long_name = "Volumetric Soil Moisture from C-band" ;
		soil_moisture_c:units = "percent" ;
		soil_moisture_c:coordinates = "Latitude Longitude" ;
		soil_moisture_c:_FillValue = -32767s ;
		soil_moisture_c:scale_factor = 1.f ;
		soil_moisture_c:add_offset = 0.f ;
	short opt_depth_c(Longitude, Latitude) ;
		opt_depth_c:long_name = "Optical Depth from C-band" ;
		opt_depth_c:coordinates = "Latitude Longitude" ;
		opt_depth_c:_FillValue = -32767s ;
		opt_depth_c:scale_factor = 0.01f ;
		opt_depth_c:add_offset = 0.f ;
	short soil_moisture_x(Longitude, Latitude) ;
		soil_moisture_x:long_name = "Volumetric Soil Moisture from X-band" ;
		soil_moisture_x:units = "percent" ;
		soil_moisture_x:coordinates = "Latitude Longitude" ;
		soil_moisture_x:_FillValue = -32767s ;
		soil_moisture_x:scale_factor = 1.f ;
		soil_moisture_x:add_offset = 0.f ;
	short opt_depth_x(Longitude, Latitude) ;
		opt_depth_x:long_name = "Optical Depth from X-band" ;
		opt_depth_x:coordinates = "Latitude Longitude" ;
		opt_depth_x:_FillValue = -32767s ;
		opt_depth_x:scale_factor = 0.01f ;
		opt_depth_x:add_offset = 0.f ;
	short mask(Longitude, Latitude) ;
		mask:long_name = "Bit Mask" ;
		mask:coordinates = "Latitude Longitude" ;
	short sm_c_error(Longitude, Latitude) ;
		sm_c_error:long_name = "Uncertainty of Soil moisture in C-band" ;
		sm_c_error:units = "percent" ;
		sm_c_error:coordinates = "Latitude Longitude" ;
		sm_c_error:_FillValue = -32767s ;
		sm_c_error:scale_factor = 0.01f ;
		sm_c_error:add_offset = 0.f ;
	short sm_x_error(Longitude, Latitude) ;
		sm_x_error:long_name = "Uncertainty of Soil moisture in X-band" ;
		sm_x_error:units = "percent" ;
		sm_x_error:coordinates = "Latitude Longitude" ;
		sm_x_error:_FillValue = -32767s ;
		sm_x_error:scale_factor = 0.01f ;
		sm_x_error:add_offset = 0.f ;

// global attributes:
		:Conventions = "CF-1.0" ;


If you have access to NASA Discover, the data is available at 
/discover/nobackup/projects/lis/RS_DATA/AMSR_E_sm/LPRM.v5/

With the availability of soil moisture from different platforms such as SMAP and SMOS and the discontinuation of AMSR-E sm products, there is no need to support this part of the LIS code.
However, you are welcome to change the LDT and LIS code to be able to use the latest version of data. 

Best
Mahdi
